### ISSUE TITLE
cannot run tomotwin_tools.py extractref

### ISSUE THREAD

**Issue by Dendooventom** [MPI-Dortmund/tomotwin-cryoet#4]
Hi guys,

I am trying to extract my selected references for picking with tomotwin, but I am getting the following error:
 
```
(tomotwin) bash-4.2$ tomotwin_tools.py extractref --tomo lmla02_ts_010.mrc_10.00Apx.mrc --coords particle_picked.coords --out tomotwin/reference/ --filename particles
Traceback (most recent call last):
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/bin/tomotwin_tools.py", line 8, in <module>
    sys.exit(_main_())
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/tomotwin/tools_main.py", line 416, in _main_
    tools = get_tool_list()
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/tomotwin/tools_main.py", line 389, in get_tool_list
    from tomotwin.modules.tools.umap import UmapTool
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/tomotwin/modules/tools/umap.py", line 6, in <module>
    import cuml
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/cuml/__init__.py", line 17, in <module>
    from cuml.common.base import Base
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/cuml/common/__init__.py", line 17, in <module>
    from cuml.common.array import CumlArray
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/cuml/common/array.py", line 25, in <module>
    from cudf import DataFrame
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/cudf/__init__.py", line 71, in <module>
    from cudf.io import (
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/cudf/io/__init__.py", line 8, in <module>
    from cudf.io.orc import read_orc, read_orc_metadata, to_orc
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/cudf/io/orc.py", line 14, in <module>
    from cudf.utils.metadata import (  # type: ignore
  File "/lmb/home/tdendooven/mambaforge/envs/tomotwin/lib/python3.9/site-packages/cudf/utils/metadata/orc_column_statistics_pb2.py", line 7, in <module>
    from google.protobuf.internal import builder as _builder
ImportError: cannot import name 'builder' from 'google.protobuf.internal' (/lmb/home/tdendooven/.local/lib/python3.9/site-packages/google/protobuf/internal/__init__.py)
```

Could you perhaps let me know which is a good version of this package to use/pin the dependency to something valid? 🙂 

Many thanks for your help!
Tom

**Comment by Dendooventom** [comment]
Hi guys,

Things work fine when going from protobuf 4.21.9 to 4.21.0. 

Cheers,
Tom

**Comment by thorstenwagner** [comment]
Hi,

thanks for letting us know. Glad that you were able to solve it. We had this issue one machine as well, the solution was different.

You might have noted that python tries to find a file OUTSIDE  your TomoTwin environment, look at the path:

```
/lmb/home/tdendooven/.local/lib/python3.9/site-packages/google/protobuf/internal/__init__.py
```
We solved it by installing it into the base environment.

Its interesting that you had the 4.21.9 installed. For every setup I did it was always 3.20.1 ... because some rapids libs (cudf) are not comptatible with anything newer than 3.20.1 its typically enforced by conda.

Anyway, this issue is still quite confusing with us
